eem-sparks/src/main/java/com/nice/eem/sparkathon/AWSS3Service.java

```diff
@@ -0,0 +1,68 @@
+package com.nice.eem.sparkathon;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+public class AWSS3Service {
+
+    /**
+     * uploading a file to s3 bucket
+     */
+    public static void uploadFile(String bucketName, String keyName, String filePath) {
+        try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.putObject(bucketName, keyName, filePath);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    /**
+     * downlnoading a file from s3 bucket
+     */
+    public static void downloadFile(String bucketName, String keyName, String filePath) {
+         try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.getObject(bucketName, keyName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * listing all the files in a bucket
+     */
+
+    public static void listFiles(String bucketName) {
+        try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.listObjects(bucketName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).